base58/src/error.rs

```diff
@@ -51,6 +50,64 @@ impl From<InvalidCharacterError> for Error {
     fn from(e: InvalidCharacterError) -> Self { Self::Decode(e) }
 }
 
+impl From<IncorrectChecksumError> for Error {
+    #[inline]
+    fn from(e: IncorrectChecksumError) -> Self { Self::IncorrectChecksum(e) }
+}
+
+impl From<TooShortError> for Error {
+    #[inline]
+    fn from(e: TooShortError) -> Self { Self::TooShort(e) }
+}
+
+/// Checksum was not correct.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct IncorrectChecksumError {
+    /// The incorrect checksum.
+    pub(super) incorrect: u32,
```

Comment:
nit: Most of errors I have seen `got` and `expected` format.

Reply:
I'm happy to change all the errors across the board and just use `got`/`expected`, will leave it for another day (or month). Thanks man.